app/models/api/submission.rb

```diff
@@ -5,5 +5,13 @@ class Submission < Base
 
     # GET /submissions/:id/calculate
     custom_endpoint :calculate, on: :member, request_method: :post
+
+    def orders_count
```

Comment:
This seems like a good change. Also, it wouldn't take much to expose these counts as metadata on the API endpoint and do all the work over there.

Reply:
yep, I think that's the right approach moving forward, especially as we are expecting submissions with many thousands of rows. It may be that the frontend app never actually needs to see the individual entries to service the user flow.

Reply:
... that said, we rely on the row to get the sheet name... either way, I think we need to re-assess what the API returns for a submission. I don't believe it's going to be viable long term to depend on being able to read all the entries to show the user the status of their submission, especially as this data could be very large. Same goes for errors. If there are a large number of errors, say because they've made the same mistake on every row, then we'll have to figure out a way to limit the number of errors we show the user, or think of some way to model multiple/repeat errors in a sensible way